fix(css): document `stretch` value

OnkarRuikar opened the pull request:
- fixes https://github.com/mdn/content/issues/29887

OnkarRuikar committed:
fix(css): document `stretch` value

dipikabh commented on files/en-us/web/css/max-height/index.md:
```diff
@@ -55,6 +56,8 @@ max-height: unset;
   - : Use the available space, but not more than [max-content](/en-US/docs/Web/CSS/max-content), i.e. `min(max-content, max(min-content, stretch))`.
 - `fit-content({{cssxref("&lt;length-percentage&gt;")}})`
   - : Uses the `fit-content` formula with the available space replaced by the specified argument, i.e. `min(max-content, max(min-content, argument))`.
+- `stretch` {{Experimental_Inline}}
+  - : Limit height of the [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the size of the element's [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). The `stretch` behaves as `100%` but it applies the resulting size to its margin box instead of the box indicated by [box-sizing](/en-US/docs/Web/CSS/box-sizing).
```
Few nits for added clarity. Plus added an additional bit to the start of the second sentence to hopefully make the second part clearer.
```suggestion
- `stretch` {{Experimental_Inline}}
  - : Limits the maximum height of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the height of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size to the margin box rather than the box determined by [box-sizing](/en-US/docs/Web/CSS/box-sizing).
```

dipikabh replied:
Similar changes can be applied to `stretch` descriptions on other pages

dipikabh commented on files/en-us/web/css/height/index.md:
```diff
@@ -59,6 +60,8 @@ height: unset;
 - `fit-content({{cssxref("&lt;length-percentage&gt;")}})`
   - : Uses the fit-content formula with the available space replaced by the specified argument, i.e. `min(max-content, max(min-content, <length-percentage>))`.
+- `stretch` {{Experimental_Inline}}
+  - : Height of the [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) is matched to the size of the element's [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). The `stretch` behaves as `100%` but it applies the resulting size to its margin box instead of the box indicated by [box-sizing](/en-US/docs/Web/CSS/box-sizing).
```
similar changes as suggested for `max-height`. Plus this description should begin with "Sets..." to be be parallel with other value descriptions.

OnkarRuikar committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

OnkarRuikar committed:
update descriptions and add notes

OnkarRuikar commented:
@dipikabh as [suggested](https://github.com/mdn/content/issues/29887#issuecomment-2289294005) I've:
- removed `stretch` definitions from pages where BCD doesn't include it
- updated descriptions
- added notes that points to BCD section

dipikabh commented on files/en-us/web/css/height/index.md:
```diff
@@ -59,6 +60,12 @@ height: unset;
+  - : Limits the height of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the height of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size to the ma ...
+
+    > [!NOTE]
+    > Refer the [browser compatiblity section](#browser_compatibility) to know about the aliases used by the browsers and the current status of the implementation of the `stretch` property.
```
Looks good, thanks for adding 👍
Added suggestion to make more compact and to use "value" instead of "property".
Uh - and on second thoughts, might be better to flip it to first say the _what_ and then give the link. (also makes "aliases" the first thing readers will notice)
```suggestion
    > To check aliases used by browsers for the `stretch` value and its implementation status, see the [Browser compatibility](#browser_compatibility) section.
```

dipikabh commented on files/en-us/web/css/block-size/index.md:
```diff
@@ -31,6 +31,7 @@ block-size: min-content;
 block-size: fit-content;
 block-size: fit-content(20em);
 block-size: auto;
+block-size: stretch;
```
should be removed

dipikabh replied:
Is this is a GitHub issue because I do see that `block-size` is not in https://github.com/mdn/content/pull/35262/commits/d3de1204ee59531013df884d951bf13ee10c7aaa as intended

OnkarRuikar replied:
An individual commit is compared with the previous commit. The files tab compares with the PR base branch, i.e., net changes.

dipikabh commented on files/en-us/web/css/inline-size/index.md:
```diff
@@ -31,6 +31,7 @@ inline-size: min-content;
 inline-size: fit-content;
 inline-size: fit-content(20em);
 inline-size: auto;
+inline-size: stretch;
```
same, not in https://github.com/mdn/content/pull/35262/commits/d3de1204ee59531013df884d951bf13ee10c7aaa as intended but showing up here

OnkarRuikar replied:
```suggestion
```

dipikabh commented on files/en-us/web/css/max-height/index.md:
```diff
@@ -57,7 +57,7 @@ max-height: unset;
   - : Uses the `fit-content` formula with the available space replaced by the specified argument, i.e. `min(max-content, max(min-content, argument))`.
 - `stretch` {{Experimental_Inline}}
-  - : Limit height of the [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the size of the element's [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). The `stretch` behaves as `100%` but it applies the resulting size to its margin box instead of the box indicated by [box-sizing](/en-US/docs/Web/CSS/box-sizing).
+  - : Limits the maximum height of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the height of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size t ...
```
The Note needs to be added here

dipikabh commented on files/en-us/web/css/max-inline-size/index.md:
```diff
@@ -31,6 +31,7 @@ max-inline-size: max-content;
 max-inline-size: min-content;
 max-inline-size: fit-content;
 max-inline-size: fit-content(20em);
+max-inline-size: stretch;
```
removed as intended in https://github.com/mdn/content/pull/35262/commits/d3de1204ee59531013df884d951bf13ee10c7aaa

OnkarRuikar replied:
```suggestion
```

dipikabh commented on files/en-us/web/css/max-width/index.md:
```diff
@@ -55,6 +56,12 @@ max-width: unset;
+  - : Limits the maximum width of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the width of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size to  ...
+
+    > [!NOTE]
+    > Refer the [browser compatiblity section](#browser_compatibility) to know about the aliases used by the browsers and the current status of the implementation of the `stretch` property.
```
changes as suggested in `height`

OnkarRuikar replied:
```suggestion
    > To check aliases used by browsers for the `stretch` value and its implementation status, see the [Browser compatibility](#browser_compatibility) section.
```

dipikabh commented on files/en-us/web/css/min-height/index.md:
```diff
@@ -56,7 +56,11 @@ min-height: unset;
+  - : Limits the minimum height of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the height of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size t ...
+
+    > [!NOTE]
+    > Refer the [browser compatiblity section](#browser_compatibility) to know about the aliases used by the browsers and the current status of the implementation of the `stretch` property.
```
as in `height`

dipikabh commented on files/en-us/web/css/min-width/index.md:
```diff
@@ -58,6 +59,12 @@ min-width: unset;
+  - : Limits the minimum width of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the width of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size to  ...
+
+    > [!NOTE]
+    > Refer the [browser compatiblity section](#browser_compatibility) to know about the aliases used by the browsers and the current status of the implementation of the `stretch` property.
```
to be updated

OnkarRuikar replied:
```suggestion
    > To check aliases used by browsers for the `stretch` value and its implementation status, see the [Browser compatibility](#browser_compatibility) section.
```

dipikabh commented on files/en-us/web/css/width/index.md:
```diff
@@ -59,6 +60,12 @@ width: unset;
+  - : Limits the width of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the width of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size to the marg ...
+
+    > [!NOTE]
+    > Refer the [browser compatiblity section](#browser_compatibility) to know about the aliases used by the browsers and the current status of the implementation of the `stretch` property.
```
to be updated

OnkarRuikar replied:
```suggestion
    > To check aliases used by browsers for the `stretch` value and its implementation status, see the [Browser compatibility](#browser_compatibility) section.
```

dipikabh commented on files/en-us/web/css/width/index.md:
```diff
@@ -59,6 +60,12 @@ width: unset;
   - : Uses the fit-content formula with the available space replaced by the specified argument, i.e. `min(max-content, max(min-content, <length-percentage>))`.
+- `stretch` {{Experimental_Inline}}
+
+  - : Limits the width of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the width of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size to the marg ...
```
```suggestion
  - : Sets the width of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the width of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size to the margin box rather than the box determined by [box-sizing](/en-US/docs/Web/CSS/box-sizing).
```

OnkarRuikar commented on files/en-us/web/css/block-size/index.md:
```diff
@@ -31,6 +31,7 @@ block-size: min-content;
 block-size: fit-content;
 block-size: fit-content(20em);
 block-size: auto;
+block-size: stretch;
```
```suggestion
```

OnkarRuikar commented on files/en-us/web/css/max-height/index.md:
```diff
@@ -55,6 +56,8 @@ max-height: unset;
 - `fit-content({{cssxref("&lt;length-percentage&gt;")}})`
   - : Uses the `fit-content` formula with the available space replaced by the specified argument, i.e. `min(max-content, max(min-content, argument))`.
+- `stretch` {{Experimental_Inline}}
+  - : Limits the maximum height of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the height of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size t ...
```
```suggestion
  - : Limits the maximum height of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the height of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size to the margin box rather than the box determined by [box-sizing](/en-US/docs/Web/CSS/box-sizing).

    > [!NOTE]
    > To check aliases used by browsers for the `stretch` value and its implementation status, see the [Browser compatibility](#browser_compatibility) section.
```

OnkarRuikar commented on files/en-us/web/css/min-block-size/index.md:
```diff
@@ -29,6 +29,7 @@ min-block-size: max-content;
 min-block-size: min-content;
 min-block-size: fit-content;
 min-block-size: fit-content(20em);
+min-block-size: stretch;
```
```suggestion
```

OnkarRuikar replied:
```suggestion
```

OnkarRuikar commented on files/en-us/web/css/min-height/index.md:
```diff
@@ -54,6 +55,12 @@ min-height: unset;
+  - : Limits the minimum height of the element's [margin box](/en-US/docs/Learn/CSS/Building_blocks/The_box_model#parts_of_a_box) to the height of its [containing block](/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block). It attempts to make the margin box fill the available space in the containing block, so in a way behaving similar to `100%` but applying the resulting size t ...
+
+    > [!NOTE]
+    > Refer the [browser compatiblity section](#browser_compatibility) to know about the aliases used by the browsers and the current status of the implementation of the `stretch` property.
```
```suggestion
    > To check aliases used by browsers for the `stretch` value and its implementation status, see the [Browser compatibility](#browser_compatibility) section.
```

OnkarRuikar replied:
```suggestion
    > To check aliases used by browsers for the `stretch` value and its implementation status, see the [Browser compatibility](#browser_compatibility) section.
```

OnkarRuikar committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

OnkarRuikar committed:
Update files/en-us/web/css/max-height/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

OnkarRuikar commented on files/en-us/web/css/min-inline-size/index.md:
```diff
@@ -27,6 +27,7 @@ min-inline-size: max-content;
 min-inline-size: min-content;
 min-inline-size: fit-content;
 min-inline-size: fit-content(20em);
+min-inline-size: stretch;
```
```suggestion
```

OnkarRuikar committed:
Apply suggestions from code review

dipikabh reviewed (approved):
Thanks a lot, @OnkarRuikar 🎉
